src/scene/axis.js

```diff
@@ -407,19 +417,52 @@ function axisTicksExtend(orient, ticks, oldScale, newScale, size, offset) {
   } else {
     size = {value: sign * size, offset: offset};
   }
+
+  var tickPlacement = config_axis.tickPlacement;
```

Comment:
@emilygu: Could you explain to me what this block of code is doing? I'm trying to understand why a particular axis needs to check other registered scales?